server/proxy/pf_context.h

```diff
@@ -35,52 +35,45 @@
  * of the connection. This lets one side of the proxy to forward events to the
  * other.
  */
-struct proxy_context
-{
-	/* Context of client or server */
-	rdpContext _context;
-
-	rdpProxyServer* server;
 
-	/**
-	 * Context of peer to which the proxy is connected.
-	 * Events from the first context are forwarded to this one.
-	 */
-	rdpContext* peerContext;
-
-	HANDLE connectionClosed;
-};
-typedef struct proxy_context proxyContext;
+typedef struct proxy_data proxyData;
 
-/**
- * Context used for the client's connection to the proxy.
- */
-struct client_to_proxy_context
+struct pserver_context
 {
-	proxyContext _context;
+	rdpContext _context;
+
+	proxyData* pdata;
 
 	HANDLE vcm;
 	HANDLE thread;
 
 	RdpgfxServerContext* gfx;
 };
-typedef struct client_to_proxy_context clientToProxyContext;
-
-BOOL init_client_to_proxy_context(freerdp_peer* client);
+typedef struct pserver_context pServerContext;
 
-/**
- * Context used for the proxy's connection to the target server.
- */
-struct proxy_to_server_context
+struct pclient_context
 {
-	proxyContext _context;
+	rdpContext _context;
+
+	proxyData* pdata;
 
 	RdpeiClientContext* rdpei;
 	RdpgfxClientContext* gfx;
 	EncomspClientContext* encomsp;
 };
-typedef struct proxy_to_server_context proxyToServerContext;
+typedef struct pclient_context pClientContext;
+
+struct proxy_data
+{
+	proxyConfig* config;
 
+	pServerContext* ps;
+	pClientContext* pc;
+
+	HANDLE connectionClosed;
+};
+
+BOOL init_client_to_proxy_context(freerdp_peer* client);
```

Comment:
Should probably be `init_pserver_context` / `init_p_server_context`

Reply:
the function names can be `p_server_context` i think